File: datasets/cpp-keep-code_davinci_001_temp_0.2/HumanEval_8_sum_product.cpp
#include<assert.h>
#include<bits/stdc++.h>
// For a given list of integers, return a tuple consisting of a sum and a product of all the integers in a list.
// Empty sum should be equal to 0 and empty product should be equal to 1.
// >>> sum_product([])
// (0, 1)
// >>> sum_product([1, 2, 3, 4])
// (10, 24)
std::tuple<long, long> sum_product(std::vector<long> numbers) {
    long sum = 0;
    long product = 1;
    for (long i : numbers) {
        sum += i;
        product *= i;
    }
    return std::make_tuple(sum, product);

}
int main() {
    auto candidate = sum_product;
    assert(candidate((std::vector<long>())) == (std::make_tuple(0, 1)));
    assert(candidate((std::vector<long>({(long)1, (long)1, (long)1}))) == (std::make_tuple(3, 1)));
    assert(candidate((std::vector<long>({(long)100, (long)0}))) == (std::make_tuple(100, 0)));
    assert(candidate((std::vector<long>({(long)3, (long)5, (long)7}))) == (std::make_tuple(15, 105)));
    assert(candidate((std::vector<long>({(long)10}))) == (std::make_tuple(10, 10)));
}

File: datasets/scala-keep-code_davinci_001_temp_0.2/HumanEval_8_sum_product.scala
import scala.math._
import scala.collection.mutable._
object Problem {
    // For a given list of integers, return a tuple consisting of a sum and a product of all the integers in a list.
    // Empty sum should be equal to 0 and empty product should be equal to 1.
    // >>> sum_product([])
    // (0, 1)
    // >>> sum_product([1, 2, 3, 4])
    // (10, 24)
    def sumProduct(numbers : List[Int]) : Tuple2[Int, Int] = {
        var sum = 0
        var product = 1
        for (num <- numbers) {
            sum += num
            product *= num
        }
        (sum, product)

    }
    def main(args: Array[String]) = {
    assert(sumProduct((List[Int]())).equals(((0, 1))));
    assert(sumProduct((List[Int](1.toInt, 1.toInt, 1.toInt))).equals(((3, 1))));
    assert(sumProduct((List[Int](100.toInt, 0.toInt))).equals(((100, 0))));
    assert(sumProduct((List[Int](3.toInt, 5.toInt, 7.toInt))).equals(((15, 105))));
    assert(sumProduct((List[Int](10.toInt))).equals(((10, 10))));
    }

}
